sql-db/hibernate-reactive/src/main/java/io/quarkus/ts/reactive/GroundedEndpoint.java

```diff
@@ -0,0 +1,94 @@
+package io.quarkus.ts.reactive;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.hibernate.reactive.mutiny.Mutiny;
+
+import io.quarkus.arc.Arc;
+import io.smallrye.mutiny.Uni;
+
+@Path("/hibernate")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class GroundedEndpoint {
+
+    Mutiny.SessionFactory factory;
+
+    public GroundedEndpoint() {
+        factory = Arc.container().instance(Mutiny.SessionFactory.class).get(); //TODO for some reason, @Inject fails. Need to investigate
+    }
+
+    @GET
+    @Path("books/{id}")
+    public Uni<Response> openSession(Integer id) {
+        return factory.openSession()
+                .flatMap(session -> session.find(Book.class, id)
+                        .map(book -> book == null
+                                ? Response.status(Response.Status.NOT_FOUND)
+                                : Response.ok(book.getTitle()))
+                        .map(Response.ResponseBuilder::build)
+                        .eventually(session::close));
+    }
+
+    @GET
+    @Path("books/author/{authorId}")
+    public Uni<Response> findAll(Integer authorId) {
+        return factory.withSession(session -> {
+            return session.createNativeQuery("Select * from books where author='" + authorId + "'", Book.class)
+                    .setMaxResults(1)
+                    .getResultList()
+                    .map(books -> Response.ok(books).build());
+        });
+    }
+
+    @GET
+    @Path("books/starts_with/{prefix}")
+    public Uni<Response> findNamed(String prefix) {
+        return factory.withSession(session -> {
+            return session.createNamedQuery("title_prefix", Book.class)
+                    .setParameter("prefix", prefix + "%")
+                    .getResultList()
+                    .map(books -> Response.ok(books).build());
+        });
+    }
+
+    @POST
+    @Path("author/create/{name}")
+    public Uni<Response> createStateless(String name) {
+        return factory.openStatelessSession().flatMap(session -> {
+            Author author = new Author();
+            author.setName(name);
+            return session.insert(author);
+        })
+                .map(ignored -> Response.status(Response.Status.CREATED))
+                .map(Response.ResponseBuilder::build);
+    }
+
+    @PUT
+    @Path("books/{authorName}/{name}")
```

Comment:
this is not the best endpoint specification IMO. replace it with URL parameters if possible

Reply:
why @PUT if it's a create method? Should be @POST

Reply:
It was meant to solve a strange behaviour of PUT endpoints in Quarkus. Changed.